src/components/modal-window/index.jsx

```diff
@@ -1,128 +1,92 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useModal } from '../context/modal-context';
 import './ModalWindow.css';
 import stubs from '../movie-list-container/mockedMovies.json';
-import { SuccessModal } from '../success-modal';
 
-export class ModalWindow extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            id: this.props.id ? this.props.id : '',
-            title: '',
-            releseDate: '',
-            movieUrl: '',
-            rating: '',
-            runtime: '',
-            overview: '',
-            showAddedMovieModal: false,
-            modalHeight: 0
-        };
-        this.handleInputChange = this.handleInputChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-        this.handleModalClose = this.handleModalClose.bind(this);
-    }
+export const ModalWindow = (props) => {
+    const [modalObject, setModalObject] = useModal();
+    const[id, setId]= useState(props.id ? props.id : 0);
+    const[title, setTitle]= useState('');
+    const[releseDate, setReleseDate]= useState('');
+    const[movieUrl, setMovieUrl]= useState('');
+    const[rating, setRating]= useState('');
+    const[runtime, setRuntime]= useState('');
+    const[overview, setOverview]= useState('');
 
-    componentDidMount(){
-        if(this.state.id !== ''){
-            const movie = stubs.filter(movie => movie.id === this.props.id)[0];
-            this.setState({
-                title: movie.title,
-                releseDate: movie.release_date,
-                movieUrl: movie.movieUrl ? movie.movieUrl : '',
-                rating: movie.vote_average,
-                runtime: movie.runtime,
-                overview: movie.overview
-            });
+    useEffect(() => {
+        if(id !== 0){
+            const movie = stubs.filter(movie => movie.id === id)[0];
+            setTitle(movie.title);
+            setReleseDate(movie.release_date);
+            setMovieUrl(() => movie.movieUrl ? movie.movieUrl : '');
+            setRating(movie.vote_average);
+            setRuntime(movie.runtime);
+            setOverview(movie.overview);
         }
-    }
-
-    handleInputChange(event) {
-        const target = event.target;
-        const value = target.value;
-        const name = target.name;
-    
-        this.setState({
-          [name]: value
-        });
-    }
+    })
```

Comment:
```suggestion
    }, [id])
```
You want to change movie whenever something changes... something like `id`. In that case, we have to use `useEffect` with `id` as a dependency. Thus, `useEffect` will be called only if `id` is changed.